packages/angular_devkit/build_angular/BUILD.bazel

```diff
@@ -194,6 +195,7 @@ ts_library(
     deps = [
         ":build_angular",
         "//packages/angular_devkit/architect/testing",
+        "//packages/angular_devkit/build_optimizer/third_party/github.com/Microsoft/TypeScript",
```

Comment:
This doesn't seem quite right, but I was getting:

```ts
ERROR: /home/circleci/ng/packages/angular_devkit/build_angular/BUILD.bazel:183:1: Compiling TypeScript (devmode) //packages/angular_devkit/build_angular:build_angular_test_lib failed (Exit 1)
packages/angular_devkit/build_angular/src/utils/build-browser-features_spec.ts:12:30 - error TS2307: [strictDeps] transitive dependency on packages/angular_devkit/build_optimizer/third_party/github.com/Microsoft/TypeScript/lib/typescript.d.ts not allowed. Please add the BUILD target to your rule's deps.

12 import { ScriptTarget } from 'typescript';
                                ~~~~~~~~~~~~
```

@gregmagolan / @filipesilva silva have you seen this in the past when vendoring packages?

Reply:
Unsure what that means, but I agree it doesn't look right. Looks like a bad resolution to the transitive one.

Reply:
I think the dep on `"//packages/angular_devkit/build_optimizer/third_party/github.com/Microsoft/TypeScript",` is winning over the `@npm//typescript` dep here. You have both listed and they are both resolvable by `typescript`. Intentional?

As for the strict deps warning, that is an odd one. I wonder if ts_library is not doing the right thing with `typescript.d.ts` in ```ts_library(
    name = "TypeScript",
    srcs = ["lib/typescript.d.ts"],
    data = ["lib/typescript.js"],
    module_name = "typescript",
    module_root = "lib/typescript.d.ts",
    visibility = [
        "//packages/angular_devkit/build_angular:__subpackages__",
        "//packages/angular_devkit/build_optimizer:__subpackages__",
    ],
)```?

Reply:
Nvrmind. I thought you were getting the error with the vendored dep there. It makes sense that you'd get it without the dep there. The transiitive module_name from a dep that depends on the vendored typescript is winning over on the import and you're getting the vendored typescript instead of the one you want. 

Since you're importing the vendored typescript with relative imports, if you remove the module_name and module_root from the its ts_library target then you won't inadvertantly resolve to it in the spec.

Reply:
Thanks @gregmagolan, that works like a charm :)